<!--
Author SiQing MA
Created in 2018-9-21 Update in 2018-9-21
This is a test version used in real turtlebot, you can't test this launch file in a simulation node!
-->

<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find Multi-Turtlebot3-Synergy)/map/Software_Museum.yaml"/>
  <arg name="open_rviz" default="false"/>

<!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)" respawn="false">
	 <param name="frame_id" value="map"/>
  </node>

<!-- BEGIN ROBOT 0 -->
	<group ns ="tb3_0">
	  <param name="tf_prefix" value="tb3_0" />
	  <!-- Turtlebot3 -->
	  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
		<arg name="model" value="$(arg model)" />
		<arg name="multi_robot_name" value="tb3_0"/>
	  </include>

	  <!-- AMCL -->
	  <node pkg="amcl" type="amcl" name="amcl" respawn="false" output="screen">
		  <remap from="map" to="/map" />
		  <param name="global_frame_id" value="map"/>
		  <param name="odom_frame_id" value="tb3_0/odom" />
		  <param name="base_frame_id" value="tb3_0/base_footprint" />
		  <param name="use_map_topic" value="true"/>
	  </node> 

	  <!-- move_base -->
	  <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
			 <remap from="map" to="/map" />
      		<param name="controller_frequency" value="10.0" />
			<param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
      		<rosparam file="$(find Multi-Turtlebot3-Synergy)/param/costmap_common_params_burger.yaml" command="load" ns="global_costmap" />
     		 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/costmap_common_params_burger.yaml" command="load" ns="local_costmap" />
     		 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/local_costmap_params.yaml" command="load" />
     		 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/global_costmap_params.yaml" command="load" />
     		 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/base_local_planner_params.yaml" command="load" />
			 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/dwa_local_planner_params_burger.yaml" command="load" />
	  </node>
	</group>
<!-- END ROBOT 0 -->

<!-- BEGIN ROBOT 1 -->
	<group ns ="tb3_1">
	  <param name="tf_prefix" value="tb3_1" />
	  <!-- Turtlebot3 -->
	  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
		<arg name="model" value="$(arg model)" />
		<arg name="multi_robot_name" value="tb3_1"/>
	  </include>

	  <!-- AMCL -->
	  <node pkg="amcl" type="amcl" name="amcl" respawn="false" output="screen">
		  <remap from="map" to="/map" />
		  <param name="global_frame_id" value="map"/>
		  <param name="odom_frame_id" value="tb3_1/odom" />
		  <param name="base_frame_id" value="tb3_1/base_footprint" />
		  <param name="use_map_topic" value="true"/>
	  </node> 

	  <!-- move_base -->
	  <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      		
			 <remap from="map" to="/map" />
      		<param name="controller_frequency" value="10.0" />
			<param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
      		<rosparam file="$(find Multi-Turtlebot3-Synergy)/param/costmap_common_params_burger.yaml" command="load" ns="global_costmap" />
     		 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/costmap_common_params_burger.yaml" command="load" ns="local_costmap" />
     		 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/local_costmap_params.yaml" command="load" />
     		 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/global_costmap_params.yaml" command="load" />
     		 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/base_local_planner_params.yaml" command="load" />
			 <rosparam file="$(find Multi-Turtlebot3-Synergy)/param/dwa_local_planner_params_burger.yaml" command="load" />
	  </node>
	</group>
<!-- END ROBOT 1 -->


  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find Multi-Turtlebot3-Synergy)/rviz/tb3_0_navigation.rviz"/>
  </group>
</launch>
